### PR TITLE
Add Code to Close imagePicker

### PR DESCRIPTION
**Pull request by parveshneedhoo**

### CODE REVIEW

File: src/ios/SOSPicker.m [hunk 1]
```diff
@@ -280,5 +281,13 @@ -(void)assetsPickerControllerDidCancel:(GMImagePickerController *)picker
    NSLog(@"GMImagePicker: User pressed cancel button");
 }
 
+- (void) closeImagePicker:(CDVInvokedUrlCommand *)command {
+    CDVPluginResult* pluginResult = nil;
+    NSArray* emptyArray = [NSArray array];
+    pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsArray:emptyArray];
+    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+    [_imagePicker.presentingViewController dismissViewControllerAnimated:YES completion:nil];
```
Comment (zfir) [inline src/ios/SOSPicker.m]:
```suggestion
    [_imagePicker.presentingViewController dismissViewControllerAnimated:YES completion:nil];
```
Where are we getting the variable `_imagePicker`?

Comment (parveshneedhoo) [reply src/ios/SOSPicker.m]: line 86

File: src/ios/SOSPicker.h [hunk 2]
```diff
@@ -16,12 +17,14 @@
...
 - (void) requestReadPermission:(CDVInvokedUrlCommand *)command;
+- (void) closeImagePicker:(CDVInvokedUrlCommand *)command;
 
 - (UIImage*)imageByScalingNotCroppingForSize:(UIImage*)anImage toSize:(CGSize)frameSize;
 
 @property (nonatomic, assign) NSInteger width;
 @property (nonatomic, assign) NSInteger height;
 @property (nonatomic, assign) NSInteger quality;
 @property (nonatomic, assign) NSInteger outputType;
+@property (nonatomic, assign) GMImagePickerController* imagePicker;
```
Comment (HashirRajah) [inline src/ios/SOSPicker.h:28]: Can we rename this to ```imagePickerController```?

File: src/ios/SOSPicker.m (hunk 1 above)
Comment (zfir) [reply src/ios/SOSPicker.m]: use `self` instead of `_`

**Review by zfir — APPROVED** [review]

File: www/imagepicker.js [hunk 3]
```diff
@@ -32,6 +32,10 @@ ImagePicker.prototype.requestReadPermission = function(callback, failureCallback
   return cordova.exec(callback, failureCallback, "ImagePicker", "requestReadPermission", []);
 };
 
+ImagePicker.prototype.closeImagePicker = function(callback) {
+  return cordova.exec(callback, null, "ImagePicker", "closeImagePicker", []);
```
Comment (dinitri) [inline www/imagepicker.js:36]: add failure callback

File: src/ios/SOSPicker.m [hunk 4]
```diff
@@ -280,5 +281,14 @@ -(void)assetsPickerControllerDidCancel:(GMImagePickerController *)picker
    NSLog(@"GMImagePicker: User pressed cancel button");
 }
 
+- (void) closeImagePicker:(CDVInvokedUrlCommand *)command {
```
Comment (dinitri) [inline src/ios/SOSPicker.m:284]: else part

File: src/ios/SOSPicker.m [hunk 5]
```diff
@@ -280,5 +281,15 @@ -(void)assetsPickerControllerDidCancel:(GMImagePickerController *)picker
...
+  if (self.imagePicker && self.imagePicker.presentingViewController) {
+    [self.imagePicker.presentingViewController dismissViewControllerAnimated:YES completion:nil];
+    CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsBool:TRUE];
+    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+  } else {
+      CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsBool:FALSE];
+      [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+  }
+}
```
Comment (HashirRajah) [inline src/ios/SOSPicker.m]:
```suggestion
- (void) closeImagePicker:(CDVInvokedUrlCommand *)command {
  bool boolMessage = FALSE;
  if (self.imagePicker && self.imagePicker.presentingViewController) {
    [self.imagePicker.presentingViewController dismissViewControllerAnimated:YES completion:nil];
    boolMessage = TRUE;
  }
  CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsBool:boolMessage];
      [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}
```

**Review by HashirRajah — APPROVED** [review]

**Pull request merged**